blob/blob.go

```diff
@@ -61,6 +61,14 @@ func (r *Reader) Size() int64 {
 	return r.r.Attributes().Size
 }
 
+// As is an escape hatch for accessing provider-specific types.
```

Comment:
While I think it's fine to link to the design in a "footnote" paragraph, the documentation comment should explain how to use the method without needing to look at the link.

(I believe this is what you're saying in your PR comment, but I want to call it out explicitly in case I misunderstood.)

Reply:
So, given that there are 4 escape hatches (with more coming after List is added), and that's just for blob, I don't want to repeat the "here's how to use As" documentation every time; the full documentation should include examples etc. My plan was to have all of that at the linked doc, and only have this short explanation here. Do you think I should repeat all of the documentation each time? Include a bit more in what's here?

Reply:
I updated them all to point to `Bucket.As`, and added more documentation there.

I also added an example test for blob.